js/main.js

```diff
@@ -134,6 +139,37 @@ $form2.addEventListener('input', function (event) {
   }
 });
 
+var id = null;
+var url = null;
+var $message = document.querySelector('span.message');
+var $buttonResults = $form1.querySelector('button.results');
+
+$form1.addEventListener('input', function (event) {
+  if (event.target.matches('input[name="usercity"]')) {
+    $buttonResults.disabled = true;
+    clearTimeout(id);
+    id = setTimeout(function () {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', 'https://api.teleport.org/api/urban_areas/');
+      xhr.responseType = 'json';
+      xhr.addEventListener('load', function () {
+        data.cityOptions = xhr.response;
+        var items = data.cityOptions._links['ua:item'];
+        for (var j = 0; j < items.length; j++) {
+          if ($form1.elements.usercity.value === items[j].name) {
+            $message.classList.add('nonvisible');
+            url = items[j].href + 'scores/';
+            break;
+          }
+          $message.classList.remove('nonvisible');
+        }
+      });
+      xhr.send();
```

Comment:
This is a neat idea I get it, however users don't wait because they're told to, they wait because there is a loading spinner that is telling them something is loading.

So you want to remove the text that asks them to wait, and add a small loading spinner under the search bar to let them know that the app is doing something in the background.